sys/include/net/gcoap.h

```diff
@@ -253,26 +253,28 @@ extern "C" {
 #endif
 
 /**
- * @brief   Size of the buffer used to write options, other than Uri-Path, in a
- *          request
+ * @brief   Reduce payload length by this value for a request created with
+ *          gcoap_req_init()
  *
- * Accommodates Content-Format and Uri-Queries
+ * Accommodates writing Content-Format option in gcoap_finish()
  */
-#define GCOAP_REQ_OPTIONS_BUF   (40)
+#define GCOAP_REQ_OPTIONS_BUF   (4)
```

Comment:
Looking at this block of changes now, I had two thoughts that came to mind:
1. why not simply join all 3 `GCOAP_X_OPTIONS_BUF` defines so something like `GCOAP_OPTIONS_FINISH_OFFSET` (there is room for improvement on the name here...)
2. more importantly: why are these defines part of the general header? As far as I can tell, they are only used internally in the `gcoap` implementation, so shouldn't they rather be part of `gcoap.c`?

Reply:
I thought about joining them, but I didn't think it was worth it. They are used only for gcoap_finish() which will be deprecated soon in favor of coap_opt_finish(). See the [wiki page](https://github.com/RIOT-OS/RIOT/wiki/CoAP-API-Options-Update) for the plan. I'm happy to make this change though if you prefer.

I will move them to `gcoap.c`. I wasn't thinking about API vs. implementation when I originally put them in the include.

Reply:
In that case, joining the defines is indeed a wasted effort. So if you just move them as they are to `gcoap.c`, I am very happy.

Reply:
Fixed in 88755f8.